src/Sarif.WorkItemFiling/Filtering/NewResultsFilteringStrategy.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.WorkItemFiling.Filtering
+{
+    /// <summary>
+    /// A filtering strategy that selects only new results.
+    /// </summary>
+    public class NewResultsFilteringStrategy : IFilteringStrategy
+    {
+        public IList<Result> FilterResults(IEnumerable<Result> results)
+        {
+            if (results == null) { throw new ArgumentNullException(nameof(results)); }
+
+            return results
+                .Where(r => r.BaselineState == BaselineState.New)
+                .ToList();
```

Comment:
>ToList [](start = 17, length = 6)

I know that you prefer the elegance of these Linq patterns but your tendency to do so (and to create Lists) tends to create performance problems that we have to solve later. Can we possibly think now about a design pattern that minimizes allocations? One approach, for example, might simply store a list of indices into a root file (rather than copying our very large result objects). A second approach might be to only refer to the filter on an as needed basis when performing grouping/refactoring (to see whether it's a candidate to be included in the grouping strategy).

Reply:
We're not copying `Result` objects; they're reference types. We're just copying references. (Will think more about this comment and respond in more detail later.)

---
In reply to: [306119043](https://github.com/microsoft/sarif-sdk/pull/1582#discussion_r306119043) [](ancestors = 306119043)

Reply:
You're right about that, of course, we will copy references only. Minimally you can allocate this collection with a predetermined count. Actual issue may be due to short-lived objects (maybe not for a large list, though).

---
In reply to: [306309402](https://github.com/microsoft/sarif-sdk/pull/1582#discussion_r306309402) [](ancestors = 306309402,306119043)

Reply:
I absolutely agree we should defer perf work until we have profiling data. Resolving.

---
In reply to: [306313524](https://github.com/microsoft/sarif-sdk/pull/1582#discussion_r306313524) [](ancestors = 306313524,306309402,306119043)